### PR TITLE
Use head_dim if in config for RoPE

### PR DESCRIPTION
**Pull request by suiyoubi**
# What does this PR do?

Fixes issue that when head_dim * num_attention_heads != hidden_size, the RoPE is not initialized correctly

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by ArthurZucker — COMMENTED** [review]
LGTM that looks like a regression (for llama at least) cc @gante !

**Comment by suiyoubi** [comment]
@gante could you please take a look at this PR ?

**Comment by suiyoubi** [comment]
Hi @ArthurZucker @gante , may I ask if there is any update on this ?

File: src/transformers/modeling_rope_utils.py [hunk 1]
```diff
@@ -58,7 +58,8 @@ def _compute_default_rope_parameters(
     elif config is not None:
         base = config.rope_theta
         partial_rotary_factor = config.partial_rotary_factor if hasattr(config, "partial_rotary_factor") else 1.0
-        dim = int((config.hidden_size // config.num_attention_heads) * partial_rotary_factor)
+        head_dim = config.head_dim if hasattr(config, "head_dim") else config.hidden_size // config.num_attention_heads
```
Comment (ArthurZucker) [inline src/transformers/modeling_rope_utils.py:61]:
```suggestion
        head_dim = getattr(config, "head_dim", config.hidden_size // config.num_attention_heads)
```

**Review by ArthurZucker — COMMENTED** [review]
Let's simplify and good to go otherwise 🤗

File: src/transformers/modeling_rope_utils.py (hunk 1 above)
Comment (suiyoubi) [reply src/transformers/modeling_rope_utils.py]: thanks for catching this! just fixed!

**Review by ArthurZucker — APPROVED** [review]
Cool thanks for updating

**Pull request merged**